### PR TITLE
Read stats from new tables/columns

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: app/v1/leaderboard.go [hunk 1]
```diff
@@ -25,74 +25,26 @@ type leaderboardResponse struct {
...
-
-			users_stats.ranked_score_%[1]s, users_stats.total_score_%[1]s, users_stats.playcount_%[1]s,
-			users_stats.replays_watched_%[1]s, users_stats.total_hits_%[1]s,
-			users_stats.avg_accuracy_%[1]s, users_stats.pp_%[1]s
+			user_stats.ranked_score, user_stats.total_score, user_stats.playcount,
+			user_stats.replays_watched, user_stats.total_hits,
+			user_stats.avg_accuracy, user_stats.pp
 		FROM users
-		INNER JOIN users_stats ON users_stats.id = users.id `
+		INNER JOIN user_stats ON user_stats.id = users.id `
```
Comment (tsunyoku) [inline app/v1/leaderboard.go:37]:
```suggestion
		INNER JOIN user_stats ON user_stats.user_id = users.id `
```

**Review by tsunyoku — CHANGES_REQUESTED** [review]

File: app/v1/leaderboard.go (hunk 1 above)
Comment (cmyui) [inline app/v1/leaderboard.go:37]:
```suggestion
		INNER JOIN user_stats ON user_stats.user_id = users.id `
```

File: app/v1/leaderboard.go [hunk 2]
```diff
@@ -165,14 +117,7 @@ func LeaderboardGET(md common.MethodData) common.CodeMessager {
...
 
-	var query string
-	if rx == 1 {
-		query = fmt.Sprintf(rxUserQuery+`WHERE users.id IN (?) ORDER BY rx_stats.pp_%[1]s DESC, rx_stats.ranked_score_%[1]s DESC`, m)
-	} else if rx == 2 {
-		query = fmt.Sprintf(apUserQuery+`WHERE users.id IN (?) ORDER BY ap_stats.pp_%[1]s DESC, ap_stats.ranked_score_%[1]s DESC`, m)
-	} else {
-		query = fmt.Sprintf(lbUserQuery+`WHERE users.id IN (?) ORDER BY users_stats.pp_%[1]s DESC, users_stats.ranked_score_%[1]s DESC`, m)
-	}
+	var query = lbUserQuery + `WHERE user_id IN (?) AND mode = ? ORDER BY user_stats.pp DESC, user_stats.ranked_score DESC`
```
Comment (cmyui) [inline app/v1/leaderboard.go:120]:
```suggestion
	var query = lbUserQuery + `WHERE user.id IN (?) AND mode = ? ORDER BY user_stats.pp DESC, user_stats.ranked_score DESC`
```

File: app/v1/leaderboard.go [hunk 3]
```diff
@@ -165,14 +117,7 @@ func LeaderboardGET(md common.MethodData) common.CodeMessager {
...
 	query, params, _ := sqlx.In(query, results)
```
Comment (cmyui) [inline app/v1/leaderboard.go:121]: not passing mode

Comment (cmyui) [reply app/v1/leaderboard.go:121]: make sure to remember rx

File: app/v1/scoreleaderboard.go [hunk 4]
```diff
@@ -108,12 +72,7 @@ func SLeaderboardGET(md common.MethodData) common.CodeMessager {
...
 	}
 
-	query := fmt.Sprintf(lbUserQuery+` ORDER BY users_stats.ranked_score_%[1]s DESC`, m)
-	if rx == 1 {
-		query = fmt.Sprintf(rxUserQuery+` ORDER BY rx_stats.ranked_score_%[1]s DESC`, m)
-	} else if rx == 2 {
-		query = fmt.Sprintf(rxUserQuery+` ORDER BY ap_stats.ranked_score_%[1]s DESC`, m)
-	}
+	query := lbUserQuery + ` ORDER BY user_stats.ranked_score DESC`
 	query, params, _ := sqlx.In(query, results)
```
Comment (cmyui) [inline app/v1/scoreleaderboard.go:76]: not passing mode -- and remember rx

File: app/v1/user.go [hunk 5]
```diff
@@ -220,70 +218,20 @@ type silenceInfo struct {
...
-ON users.id=users_stats.id
-LEFT JOIN rx_stats
-ON users.id=rx_stats.id
-LEFT JOIN ap_stats
-ON users.id=ap_stats.id
-WHERE ` + whereClause + ` AND ` + md.User.OnlyUserPublic(true) + `
-LIMIT 1
+	user_stats.ranked_score, user_stats.total_score, user_stats.playcount, user_stats.playtime,
+	user_stats.replays_watched, user_stats.total_hits,
+	user_stats.avg_accuracy, user_stats.pp, user_stats.max_combo,
```
Comment (cmyui) [inline app/v1/user.go:231]:
```suggestion
	user_stats.avg_accuracy, user_stats.pp, user_stats.max_combo
```

File: app/v1/user.go [hunk 6]
```diff
@@ -220,70 +218,20 @@ type silenceInfo struct {
...
+FROM user_stats
+LEFT JOIN users ON users.id = user_stats.user_id
+WHERE ` + whereClause + ` AND ` + md.User.OnlyUserPublic(true) + ` AND user_stats.mode = ?
 `
 	// Fuck.
```
Comment (cmyui) [inline app/v1/user.go:236]:
```suggestion
```

File: app/v1/user.go [hunk 7]
```diff
@@ -348,6 +264,66 @@ LIMIT 1
 		return Err500
 	}
 
+	// Scan stats into response for all gamemodes, across vn/rx/ap
+	for _, relaxMode := range []int{0, 1, 2} {
+		modeOffset := relaxMode * 4
+		// Scan vanilla gamemode information into response
+		err = md.DB.QueryRow(query, userIdParam, 0+modeOffset).Scan(
+			r.Stats[relaxMode].STD.RankedScore, &r.Stats[relaxMode].STD.TotalScore, &r.Stats[relaxMode].STD.PlayCount, &r.Stats[relaxMode].STD.PlayTime,
```
Comment (cmyui) [inline app/v1/user.go:272]:
```suggestion
			&r.Stats[relaxMode].STD.RankedScore, &r.Stats[relaxMode].STD.TotalScore, &r.Stats[relaxMode].STD.PlayCount, &r.Stats[relaxMode].STD.PlayTime,
```

File: app/v1/user.go [hunk 8]
```diff
@@ -348,6 +264,66 @@ LIMIT 1
...
+		case err == sql.ErrNoRows:
+			return common.SimpleResponse(404, "That user could not be found!")
+		case err != nil:
+			md.Err(err)
+			return Err500
+		}
+
+		// Scan taiko gamemode information into response
+		err = md.DB.QueryRow(query, userIdParam, 1+modeOffset).Scan(
+			r.Stats[relaxMode].Taiko.RankedScore, &r.Stats[relaxMode].Taiko.TotalScore, &r.Stats[relaxMode].Taiko.PlayCount, &r.Stats[relaxMode].Taiko.PlayTime,
```
Comment (cmyui) [inline app/v1/user.go:286]:
```suggestion
			&r.Stats[relaxMode].Taiko.RankedScore, &r.Stats[relaxMode].Taiko.TotalScore, &r.Stats[relaxMode].Taiko.PlayCount, &r.Stats[relaxMode].Taiko.PlayTime,
```

File: app/v1/user.go [hunk 9]
```diff
@@ -348,6 +264,66 @@ LIMIT 1
...
+		case err == sql.ErrNoRows:
+			return common.SimpleResponse(404, "That user could not be found!")
+		case err != nil:
+			md.Err(err)
+			return Err500
+		}
+
+		// Scan ctb gamemode information into response
+		err = md.DB.QueryRow(query, userIdParam, 2+modeOffset).Scan(
+			r.Stats[relaxMode].CTB.RankedScore, &r.Stats[relaxMode].CTB.TotalScore, &r.Stats[relaxMode].CTB.PlayCount, &r.Stats[relaxMode].CTB.PlayTime,
```
Comment (cmyui) [inline app/v1/user.go:300]:
```suggestion
			&r.Stats[relaxMode].CTB.RankedScore, &r.Stats[relaxMode].CTB.TotalScore, &r.Stats[relaxMode].CTB.PlayCount, &r.Stats[relaxMode].CTB.PlayTime,
```

File: app/v1/user.go [hunk 10]
```diff
@@ -348,6 +264,66 @@ LIMIT 1
...
+		case err == sql.ErrNoRows:
+			return common.SimpleResponse(404, "That user could not be found!")
+		case err != nil:
+			md.Err(err)
+			return Err500
+		}
+
+		// Scan mania gamemode information into response
+		err = md.DB.QueryRow(query, userIdParam, 3+modeOffset).Scan(
+			r.Stats[relaxMode].Mania.RankedScore, &r.Stats[relaxMode].Mania.TotalScore, &r.Stats[relaxMode].Mania.PlayCount, &r.Stats[relaxMode].Mania.PlayTime,
```
Comment (cmyui) [inline app/v1/user.go:314]:
```suggestion
			&r.Stats[relaxMode].Mania.RankedScore, &r.Stats[relaxMode].Mania.TotalScore, &r.Stats[relaxMode].Mania.PlayCount, &r.Stats[relaxMode].Mania.PlayTime,
```

File: app/v1/user.go [hunk 11]
```diff
@@ -348,6 +264,66 @@ LIMIT 1
 		return Err500
 	}
 
+	// Scan stats into response for all gamemodes, across vn/rx/ap
+	for _, relaxMode := range []int{0, 1, 2} {
```
Comment (cmyui) [inline app/v1/user.go:268]: There's a behaviour chang here in that game modes 9, 10, and 11 will be filled

Pretty sure this is fine, as 7 was being written before, and is invalid.

Atop in the response struct definition, it's set to contain all modes for all 3 rx modes.

File: app/v1/leaderboard.go [hunk 12]
```diff
@@ -165,15 +118,8 @@ func LeaderboardGET(md common.MethodData) common.CodeMessager {
...
-	var query string
-	if rx == 1 {
-		query = fmt.Sprintf(rxUserQuery+`WHERE users.id IN (?) ORDER BY rx_stats.pp_%[1]s DESC, rx_stats.ranked_score_%[1]s DESC`, m)
-	} else if rx == 2 {
-		query = fmt.Sprintf(apUserQuery+`WHERE users.id IN (?) ORDER BY ap_stats.pp_%[1]s DESC, ap_stats.ranked_score_%[1]s DESC`, m)
-	} else {
-		query = fmt.Sprintf(lbUserQuery+`WHERE users.id IN (?) ORDER BY users_stats.pp_%[1]s DESC, users_stats.ranked_score_%[1]s DESC`, m)
-	}
-	query, params, _ := sqlx.In(query, results)
+	var query = lbUserQuery + `WHERE user.id IN (?) AND mode = ? ORDER BY user_stats.pp DESC, user_stats.ranked_score DESC`
```
Comment (cmyui) [inline app/v1/leaderboard.go:121]:
```suggestion
	var query = lbUserQuery + `WHERE user.id IN (?) AND user_stats.mode = ? ORDER BY user_stats.pp DESC, user_stats.ranked_score DESC`
```

File: app/peppy/user.go [hunk 13]
```diff
@@ -29,43 +30,40 @@ func GetUser(c *fasthttp.RequestCtx, db *sqlx.DB) {
 	}
 	var user osuapi.User
 	whereClause, p := genUser(c, db)
-	whereClause = "WHERE " + whereClause
+	whereClause = "WHERE " + whereClause + " AND mode = ? "
```
Comment (cmyui) [inline app/peppy/user.go:33]:
```suggestion
	whereClause = "WHERE " + whereClause + " AND user_stats.mode = ? "
```

File: .github/workflows/production-deploy.yml [hunk 14]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - master
+      - new-stats-reads
```
Comment (cmyui) [inline .github/workflows/production-deploy.yml:7]:
```suggestion
```

**Pull request merged**